src/exiv2.rs

```diff
@@ -0,0 +1,60 @@
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use std::env;
+use std::convert::From;
+use std::error;
+use std::fmt;
+use serde_json;
+use regex::Regex;
+
+pub fn read_metadata_from_file(file_path: &Path) -> Result<serde_json::Value, ExifReadError> {
+    let exec_path = get_exiv2json_path();
+    let exiv2json_output = Command::new(exec_path)
+        .arg(file_path.to_str().unwrap())
+        .output()
+        .expect("failed to execute exiv2json process");
+
+    let output_str = String::from_utf8(exiv2json_output.stdout)?;
+    let json = remove_control_characters(&output_str);
+
+    serde_json::from_str(&json).map_err(|_err| ExifReadError {})
+}
+
+fn get_exiv2json_path() -> PathBuf {
+    let base_path = if let Ok(cargo_manifest_dir) = env::var("CARGO_MANIFEST_DIR") {
+        PathBuf::from(cargo_manifest_dir).join("release").join("exiv2")
+    } else {
+        env::current_exe().unwrap()
+            .parent().unwrap()
+            .canonicalize().unwrap() // TODO: Simplify this
+    };
+
+    base_path.join("exiv2json")
+}
+
+fn remove_control_characters(output: &str) -> String {
+    let pattern = Regex::new(r"\p{Cc}").unwrap();
```

Comment:
co to robi?

Reply:
Usuwa znaki kontrolne, czyli te nie wyświetlalne znaki, które służą do sterowania terminalem czy czymś tam co przetwarza te dane.
Te znaki generalnie przeszkadzają serde i tu je usuwam.

Reply:
+1